src/main/java/myapp/HelperBuddy/Task.java

```diff
@@ -0,0 +1,66 @@
+package myapp.helperbuddy;
+
+/**
+ * Task abstract class which is inherited from subclasses
+ */
+abstract class Task {
+    private String description;
+    private boolean isDone;
+
+    /**
+     * Constructs a Task with the specified description.
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the description of the task.
+     * @return The description of the task.
+     */
+    public String getDescription() {
+        return this.description;
+    }
+
+    /**
+     * Returns whether the task is marked as done.
+     * @return true if the task is done, false otherwise.
+     */
+    public boolean getDone() {
+        return this.isDone;
+    }
+
+    /**
+     * Marks the task as done.
+     * This method sets the completion status of the task to true.
+     */
+    public void markDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as not done.
+     * This method sets the completion status of the task to false.
+     */
+    public void markUndone() {
```

Comment:
Perhaps markDone and markUndone can be combined into something like toggleMark? using
this.isDone = !this.isDone